packages/kbn-user-profile-components/src/user_avatar.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EuiAvatarProps } from '@elastic/eui';
+import { EuiAvatar, useEuiTheme } from '@elastic/eui';
+import type { FunctionComponent, HTMLAttributes } from 'react';
+import React from 'react';
+
+import type { UserProfile, UserProfileAvatarData } from './imported_types/user_profile';
+import {
+  getUserAvatarColor,
+  getUserAvatarInitials,
+  USER_AVATAR_MAX_INITIALS,
+} from './imported_types/user_profile';
+import { getUserDisplayName } from './imported_types/user';
+
+export type UserProfileWithAvatar = UserProfile<{ avatar?: UserProfileAvatarData }>;
+
+export interface UserAvatarProps extends Omit<HTMLAttributes<HTMLDivElement>, 'color'> {
```

Comment:
Most of the props from `EuiAvatar` are not applicable to this component since they would override what is rendered which is exactly what we want to control in this convenience component. However I can definitely make this more flexible and so that consumers can pass in things like `aria-label` and `data-test-subj` which could be useful.